### PR TITLE
Make telepathy build

### PR DESCRIPTION
**Pull request by joakim-tjernlund**
A few libs where missing at link time and removed some stubs as these are now impl.

### CODE REVIEW

File: src/telepathy/telepathy-stubs.c [hunk 1]
```diff
@@ -303,19 +303,6 @@ void sipe_backend_notify_error(SIPE_UNUSED_PARAMETER struct sipe_core_public *si
 			       SIPE_UNUSED_PARAMETER const gchar *title,
 			       SIPE_UNUSED_PARAMETER const gchar *msg) {}
 
-/** USER *********************************************************************/
-
-void sipe_backend_user_feedback_typing(SIPE_UNUSED_PARAMETER struct sipe_core_public *sipe_public,
```
Comment (xhaakon) [inline src/telepathy/telepathy-stubs.c:308]: Where exactly do you see these implemented? I did a grep of the source code and this is the only place where I see the functions mentioned in `telepathy/` subdir.

Comment (joakim-tjernlund) [reply src/telepathy/telepathy-stubs.c:308]: I noticed that these symbols were doubly defined when linking so I just assumed that
the code had impl. them, sorry.

Maybe this error has been there all along but just exposed now because sharing pulls in some new code from farstream?

Comment (xhaakon) [reply src/telepathy/telepathy-stubs.c:308]: The build fails during linking you say? Hmm, maybe telepathy plugin wrongly tries to link with both telepathy and purple backends (which each define their own versions of those functions). Can you post your full build log?

Comment (joakim-tjernlund) [reply src/telepathy/telepathy-stubs.c:308]:
```
Making all in telepathy
Making all in data
libtool: link: gcc -Werror -Wall -Wextra -Waggregate-return -Wcast-align -Wdeclaration-after-statement -Wdeprecated-declarations -Winit-self -Wmaybe-uninitialized -Wmissing-declarations -Wmissing-prototypes -Wnested-externs -Wpointer-arith -Wundef -Wunused-but-set-variable -DLOCALEDIR=\"/usr/local/share/locale\" -pthread -I/usr/include/telepathy-1.0 -I/usr/include/dbus-1.0 -I/usr/lib64/dbus-1.0/include -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/dbus-1.0 -I/usr/lib64/dbus-1.0/include -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -pthread -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I./../api -g -O2 -o telepathy-sipe telepathy_sipe-telepathy-buddy.o telepathy_sipe-telepathy-connection.o telepathy_sipe-telepathy-debug.o telepathy_sipe-telepathy-dnsquery.o telepathy_sipe-telepathy-main.o telepathy_sipe-telepathy-protocol.o telepathy_sipe-telepathy-schedule.o telepathy_sipe-telepathy-search.o telepathy_sipe-telepathy-status.o telepathy_sipe-telepathy-stubs.o telepathy_sipe-telepathy-tls.o telepathy_sipe-telepathy-transport.o  ../core/.libs/libsipe_core.a -lwinpr -lfreerdp-shadow -lfreerdp-shadow-subsystem -L/usr/lib64 ../purple/.libs/libsipe_backend_vv.a ../purple/.libs/libsipe_backend.a ../core/.libs/libsipe_core_crypto.a ../core/.libs/libsipe_core_libxml2.a ../core/.libs/libsipe_core_mime.a -lgmime-2.6 -lxml2 -lssl3 -lsmime3 -lnss3 -lnssutil3 -lplds4 -lplc4 -lnspr4 -lcrypto -ltelepathy-glib -ldbus-glib-1 -ldbus-1 -lgio-2.0 -lgobject-2.0 -lglib-2.0 -lpurple -lgssapi_krb5 -lkrb5 -lk5crypto -lcom_err -pthread
../purple/.libs/libsipe_backend.a(libsipe_backend_la-purple-user.o): In function `sipe_backend_user_feedback_typing':
/usr/local/src/sipe/src/purple/purple-user.c:48: multiple definition of `sipe_backend_user_feedback_typing'
telepathy_sipe-telepathy-stubs.o:/usr/local/src/sipe/src/telepathy/telepathy-stubs.c:309: first defined here
../purple/.libs/libsipe_backend.a(libsipe_backend_la-purple-user.o): In function `sipe_backend_user_feedback_typing_stop':
/usr/local/src/sipe/src/purple/purple-user.c:57: multiple definition of `sipe_backend_user_feedback_typing_stop'
telepathy_sipe-telepathy-stubs.o:/usr/local/src/sipe/src/telepathy/telepathy-stubs.c:311: first defined here
../purple/.libs/libsipe_backend.a(libsipe_backend_la-purple-user.o): In function `sipe_backend_user_ask':
/usr/local/src/sipe/src/purple/purple-user.c:70: multiple definition of `sipe_backend_user_ask'
telepathy_sipe-telepathy-stubs.o:/usr/local/src/sipe/src/telepathy/telepathy-stubs.c:316: first defined here
../purple/.libs/libsipe_backend.a(libsipe_backend_la-purple-user.o): In function `sipe_backend_user_close_ask':
/usr/local/src/sipe/src/purple/purple-user.c:87: multiple definition of `sipe_backend_user_close_ask'
telepathy_sipe-telepathy-stubs.o:/usr/local/src/sipe/src/telepathy/telepathy-stubs.c:317: first defined here
collect2: error: ld returned 1 exit status
```

File: src/telepathy/Makefile.am [hunk 2]
```diff
@@ -35,6 +35,8 @@ telepathy_sipe_CFLAGS = \
 
 telepathy_sipe_LDADD = \
 	../core/libsipe_core.la \
+	../purple/libsipe_backend_vv.la \
+	../purple/libsipe_backend.la \
```
Comment (xhaakon) [inline src/telepathy/Makefile.am:39]: Oh, sure. I didn't actually look what you've done here. These two have no business being used with Telepathy backend and are the reason for multiple definition during linking. Why did you add them?

File: src/telepathy/Makefile.am [hunk 3]
```diff
@@ -46,4 +48,4 @@ telepathy_sipe_LDADD = \
 	$(DBUS_GLIB_LIBS) \
 	$(GIO_LIBS) \
 	$(GOBJECT_LIBS) \
-	$(GLIB_LIBS)
+	$(GLIB_LIBS) -lpurple
```
Comment (xhaakon) [inline src/telepathy/Makefile.am:51]: Same here. Either it's purple backend and links with libpurple, or Telepathy backend and links with... well, whatever telepathy extensions need to link to.

Comment (joakim-tjernlund) [reply src/telepathy/Makefile.am:51]: I just looked at the link failures I got for telepathy and looked where the syms were
defined. I guess there is more to it and I suspect the code has to be reorganized
for telepathy to build.

File: src/core/Makefile.am [hunk 4]
```diff
@@ -217,7 +217,7 @@ libsipe_core_la_SOURCES += \
 libsipe_core_la_CFLAGS += \
 	$(FREERDP_CFLAGS)
 
-libsipe_core_la_LIBADD = \
+libsipe_core_la_LIBADD = -lwinpr \
 	$(FREERDP_LIBS)
 endif
 >>>>>>> 04b3590 appshare: build optionally only when we have FreeRDP headers
```
Comment (xhaakon) [inline src/core/Makefile.am:223]: Oops, looks like I screwed this line up during a rebase...

File: src/telepathy/Makefile.am (hunk 3 above)
Comment (xhaakon) [reply src/telepathy/Makefile.am:51]: Okay, never mind. I'll anyway make changes to the build files before moving the code upstream. For now let's just keep telepathy disabled.
